component-builder/src/component_builder/build.py

```diff
@@ -83,11 +92,9 @@ def post(self, stage, components):
 def run(mode, components, status_callback=None, optional=False):
     errors = []
 
-    titles = []
     for comp in components:
-        titles.append(comp.title)
+        declare_component_usage(comp)
```

Comment:
I think just remove this and `declare_component_usage` and users can explicitly call `declare` in their jenkinsfile/circle.yaml